src/components/BioForm.js

```diff
@@ -0,0 +1,76 @@
+import React, { useEffect, useState } from 'react'
+import { useLocation, useNavigate } from 'react-router-dom';
+import bioRequest from '../services/BioRequest';
+
+const BioForm = (props) => {
+
+    const navigate = useNavigate();
+
+    const { state } = useLocation();
+
+    const [credentials, setCredentials] = useState({
+        name: '',
+        address: '',
+        description: ''
+    })
+    
+    useEffect(() => {
+        if(props.method === 'PATCH'){
+            const {data} = state;
+
+        setCredentials({
+            name: data.name,
+            address: data.address,
+            description: data.description
+        })
+    }
+
+    }, []);
```

Comment:
Can you please explain how this works?
Fix indentation!

Reply:
The update function works as a patch (partial update). The useEffect() runs once after the page has been rendered (Form has been displayed). 

The function checks if the method is patch or post (as the same code runs for the create Bio as well. 

If the method is patch (meaning the data is to be updated), the data that would have been gotten through the state props in Link component ("<Link className="method" to='/blog/edit/:id'  state={{id: props.id, data: props.data}}> Update </Link>"). This data is fetched on another page using the useLocation() hook.

The data is set to the input values of the fields to pre-fill the fields such that the user can choose to update them with the previously value autofilled. The user can leave fields empty as well as the update is a partial update (patch).

The cycle to reach an update is (BioPage --> Record Container --> Method --> Link). The Link component passes the data (as shown above) and calls the UpdateOrCreate Page. That loads the BioForm as an update form and autofills previous data in it.